Tests/SonarScanner.MSBuild.Tasks.IntegrationTests/TargetsTests/RoslynTargetsTests.cs

```diff
@@ -188,10 +204,18 @@ public BuildLog Execute_Roslyn_Settings_ValidSetup(bool isTestProject, string ms
     <AdditionalFiles Include='x:\aaa\project.additional.file.2.txt' />
 </ItemGroup>
 ";
-            var projectFilePath = CreateProjectFile(config, projectSnippet);
+
+            var projectDirectory = TestUtils.CreateTestSpecificFolderWithSubPaths(TestContext);
```

Comment:
Wasn't it the purpose of refactoring if this is the same all over and to avoid duplications ? Not sure to understand your point here.

Reply:
`CreateProjectFile` and the removed `CreateCaptureDataTargetsFile` method had around 70 lines. The great refactoring removed most of them in a reusable way and that's great. 

What was left was these 5 lines that are specific to `RoslynTargetTests.cs` file.
```
            var projectDirectory = TestUtils.CreateTestSpecificFolderWithSubPaths(TestContext);
            var targetTestUtils = new TargetsTestsUtils(TestContext);

            var projectTemplate = targetTestUtils.GetProjectTemplate(config, projectDirectory);

            var filePath = targetTestUtils.CreateProjectFile(projectDirectory, GetProjectData(projectTemplate, projectDirectory, projectSnippet));

            targetTestUtils.CreateCaptureDataTargetsFile(projectDirectory, AfterTargets);
```

I suggest to keep the method `CreateProjectFile(AnalysisConfig analysisConfig, string testSpecificProjectXml)` that would contain just these 5 lines of code. And to call this method from those 13 tests in this file to avoid duplications of these surviving 5 lines.

Same concept applies to other files in this PR that were improved by the template refactoring.